src/LSST/system/ComServer.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+ *  This file is part of LSST M2 support system package.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "system/ComServer.h"
+
+// System headers
+#include <functional>
+#include <thread>
+
+// Project headers
+#include "system/Config.h"
+#include "system/Log.h"
+
+// LSST headers
+
+using namespace std;
+using namespace std::placeholders;
+
+namespace LSST {
+namespace m2cellcpp {
+namespace system {
+
+ComServer::Ptr ComServer::create(IoContextPtr const& ioContext, int port) {
+    return ComServer::Ptr(new ComServer(ioContext, port));
+}
+
+ComServer::ComServer(IoContextPtr const& ioContext, int port)
+        : _ioContext(ioContext),
+          _port(port),
+          _acceptor(*_ioContext, boost::asio::ip::tcp::endpoint(boost::asio::ip::tcp::v4(), _port)) {
+    // Set the socket reuse option to allow recycling ports after catastrophic
+    // failures.
+    _acceptor.set_option(boost::asio::socket_base::reuse_address(true));
+}
+
+ComServer::~ComServer() {
+    Log::log(Log::DEBUG, "ComServer::~ComServer()");
+}
+
+void ComServer::run() {
+    Log::log(Log::DEBUG, "ComServer::run()");
+    // Begin accepting immediately. Otherwise it will finish when it
+    // discovers that there are outstanding operations.
+    _beginAccept();
+
+    // Launch all threads in the pool
+    int threadCount = stoi(Config::get().getValue("server", "threads"));
+    vector<shared_ptr<thread>> threads(threadCount);
```

Comment:
We will want to be able to define the number of threads available in a configuration file in a latter time.